### DISCUSSION TITLE
Using Set/Variables in SQLScript?

### DISCUSSION THREAD

**Discussion by RobertLHarris** [dbeaver/dbeaver#12594] [Q&A]
I have to create a large number of users and other maintenance that is rather repetative.  Looking around, it looks like I should be able to do this:

@set Username = '`username1`@`%`';
@set Password = 'SomeSt1ng._here!';

CREATE USER ${Username} IDENTIFIED BY ${Password};
GRANT USAGE on *.* to ${Username};
GRANT SELECT on `Table1` to ${Username};
GRANT SELECT on `Table2` to ${Username};
<and so on>

But I get syntax errors on all the variable substitutions.  I have enabled the "use variables" in the preferences already as well.

Any help?

**Comment by Hurtplug** [comment]
Hello, i'm discovering Dbeaver, it' a very clever software,

I'm using the Dbeaver variable system, but from my point of view those variable may be missing something.

This is my script :

@set db = DB01CWE5002
@set dated = 1635724800
@set datef = 1638316800
select*from ${db}.dbo.AD10Min1_Average WHERE tstamp BETWEEN ${dated} AND ${datef}; 

To be able to output one file for one table, i'm using the TASK in Dbeaver.

With the variable of Dbeaver i can change 3 variables and do hundred of select easily.

Thing is to output a file for each table, i use the TASK system, and in the task system Dbearver doesn't interpret the code like in a script.

In a TASK Dbeaver doesnt seem to read the code destined to him, it just sent it to the SQL server, who output an error.

is there anything i miss ?

nb : i only have a read access to my servers, and they are very old microsoft sql, i don't think i could use any variable server side.
